src/tracer.js

```diff
@@ -0,0 +1,73 @@
+'use strict'
+
+import BasicSpan from './span'
+import {TextMapPropagator, BinaryPropagator} from './propagation'
+import {DefaultSampler} from './sampler'
+import {DefaultRecorder} from './recorder'
+
+// Implement https://github.com/opentracing/opentracing-javascript/blob/master/src/tracer.js
+export default class BasicTracer {
+    /**
+     * Constructs basic tracer instance with provided Sampler and Recorder
+     * @param  {Object} options
+     *         Optional associative array of fields.
+     *         - `sampler` {Sampler} Optional object with `isSample` method, the
+     *             method provided with current span as arguments, return
+     *             Boolean value indicate whether should take current span
+     *             as sample. See src/sample.js for example.
+     *         - `recorder` {Recorder} Optional object with `record` method, the
+     *             method take span and do whatever required to record a span.
+     *             See src/recorder.js for example.
+     *
+     * @return {BasicTracer}
+     */
+    constructor({sampler, recorder} = {}) {
+        this._sampler = sampler || new DefaultSampler()
+        this._recorder = recorder || new DefaultRecorder()
+        this._binaryPropagator = new BinaryPropagator(this)
+        this._textPropagator = new TextMapPropagator(this)
+    }
+
+    setInterface(inf) {
+        this._interface = inf
+    }
+
+    startSpan(fields) {
+        // Interface or Implementation argument
+        // https://github.com/opentracing/opentracing-javascript/pull/29
```

Comment:
it didn't look like https://github.com/opentracing/opentracing-javascript/pull/29 was going to go through... isn't it safe to assume that `fields.parent` is an opentracing `Span` and thus has an `imp()` method?